### DISCUSSION TITLE
Local folder scraping for new images to upload?

### DISCUSSION THREAD

**Discussion by wilsonics** [rr-/szurubooru#665] [Q&A]
I don't know if anyone can help me out with this one, but I'm looking for a way to have szurubooru (or a bash script) look in a particular directory and then post them to the site.  I'd also like to see if there's a way to fetch EXIF data from the images to set the tags.

A little background on this: I use stable-diffusion to create images and all the prompts are stored in EXIF data of the jpg.  Instead of me going into the site's upload portion of the webui manually, I'd like to see if I can automate all the data in EXIF and translate to suzubooru tags, and make a new post. 

Thanks in advance!

**Comment by nocrcl** [comment]
Hey :3

i wrote a bash script that might help.
[ul2szuru.sh](https://github.com/nocrcl/ul2szuru/tree/main)

It is not automated, __yet__!
But it already works when fired manually. With a find command (example in the README) you can scrape through folders and subfolders.

It works on files generated with/by dezgo.com. Write the prompt like comma separated tags, and no need for underscores between two or more word tags.
Enter your prompt like "ren and stimpy, hiking, mountains, grassy hills" and it creates the tags: 
```
ren_and_stimpy
hiking
mountains
grassy_hills
```

Oh, and it still depends on exiv2 for getting the image metadata and jq to generate the tags in json format.

**Comment by neobooru** [comment]
[reluce/szurubooru-toolkit](https://github.com/reluce/szurubooru-toolkit) might be usable for you.

Or you can also create your own script to do this. I would recommend using python for this. I don't think anyone has made a standalone python library to interact with szurubooru, but you can use [this code](https://github.com/rr-/szurubooru/discussions/570#discussioncomment-6015822) as a starting point for uploading images. [Fernsicles/szurutagger](https://github.com/Fernsicles/szurutagger) might also be interesting for you, mostly to look at the szuruAPI.py file. Oh and the [API docs are here.](https://github.com/rr-/szurubooru/blob/master/doc/API.md#creating-post)

**Reply by wilsonics** [reply]
Thanks very much, I'll give that a shot.
